renku/api/datasets.py

```diff
@@ -449,6 +451,65 @@ def get_relative_url(self, url):
             url = str(url / submodule_url.name)
         return url
 
+    def get_newest_commit_for_dataset(self, dataset):
+        """Gets the newest commit for a dataset or its files."""
+        paths = [(Path(dataset.path) / self.METADATA).resolve()]
+
+        paths.extend(f.full_path for f in dataset.files)
+
+        commit = list(self.repo.iter_commits(max_count=1, paths=paths))[0]
+
+        return commit
+
+    def add_dataset_tag(self, dataset, tag, description='', force=False):
+        """Adds a new tag to a dataset.
+
+        Validates if the tag already exists and that the tag follows
+        the same rules as docker tags.
+        See https://docs.docker.com/engine/reference/commandline/tag/
+        for a documentation of docker tag syntax.
+        """
```

Comment:
Can we also add ``:raises: ValueError ``? Adding those in docstrings is a good practice we should enforce since it's much easier to reason about the function by just reading the docstring. (Same goes for other places in the PR)